lib/Extension/CodeTransformExtra/Rpc/GenerateAccessorHandler.php

```diff
@@ -33,27 +46,86 @@ public function name(): string
 
     public function configure(Resolver $resolver)
     {
+        $resolver->setDefaults([
+            self::PARAM_OFFSET => null,
+            self::PARAM_NAMES => null,
+        ]);
         $resolver->setRequired([
             self::PARAM_PATH,
             self::PARAM_SOURCE,
-            self::PARAM_OFFSET,
         ]);
     }
 
     public function handle(array $arguments)
     {
-        $sourceCode = $this->generateAccessor->generateAccessor(
-            SourceCode::fromStringAndPath(
-                $arguments[self::PARAM_SOURCE],
-                $arguments[self::PARAM_PATH]
-            ),
-            $arguments[self::PARAM_OFFSET]
-        );
+        $class = $this->class($arguments[self::PARAM_SOURCE]);
+
+        if (null === $arguments[self::PARAM_OFFSET]) {
+            $this->requireInput(ListInput::fromNameLabelChoices(
+                self::PARAM_NAMES,
+                sprintf('Properties from "%s"', $class->name()),
+                $this->propertiesChoices($class)
+            )->withMultiple(true));
+        }
+
+        if ($this->hasMissingArguments($arguments)) {
+            return $this->createInputCallback($arguments);
+        }
+
+        $originalSource = $arguments[self::PARAM_SOURCE];
+        $newSource = SourceCode::fromStringAndPath($originalSource, $arguments[self::PARAM_PATH]);
+
+        if ($offset = $arguments[self::PARAM_OFFSET]) {
+            $newSource = $this->generateAccessor->generateFromOffset($newSource, $offset);
+        } else {
+            foreach ($arguments[self::PARAM_NAMES] as $propertyName) {
+                $newSource = $this->generateAccessor->generateFromPropertyName($newSource, $propertyName);
+            }
+        }
 
         return UpdateFileSourceResponse::fromPathOldAndNewSource(
-            $sourceCode->path(),
-            $arguments[self::PARAM_SOURCE],
-            (string) $sourceCode
+            $newSource->path(),
+            $originalSource,
+            (string) $newSource
         );
     }
+
+    /**
+     * @param string $source
+     *
+     * @return ReflectionClass
+     */
+    private function class($source)
+    {
+        $classes = $this->reflector->reflectClassesIn($source);
+
+        if ($classes->count() === 0) {
+            throw new InvalidArgumentException(
+                'No classes in source file'
+            );
+        }
+
+        if ($classes->count() > 1) {
+            throw new InvalidArgumentException(
+                'Currently will only generates accessor by name in files with one class'
+            );
+        }
+
+        return $classes->first();
+    }
+
+    private function propertiesChoices(ReflectionClass $class): array
+    {
+        // Select only those from the current class because the accessor generator
+        // is not able to work with the parent class at the time
+        $properties = $class->properties()->belongingTo($class->name());
```

Comment:
:+1: Typically I can't think of situations where I want to generate an accessor for a parent class.

Reply:
I had the same reaction when I wrote it but I didn't gave it more thought since the generator was not able to do it.
In general it should be a bad idea to gave access to a parent attribute anyway.

Sorry for the wait, I didn't saw you reviews last time...